File: docs/api/spring/com/shoptourr/api/v1/dto/stats/StatsDtos.java
package com.shoptourr.api.v1.dto.stats;

import com.shoptourr.api.v1.dto.common.CommonDtos.MoneyDto;
import com.shoptourr.api.v1.dto.purchase.PurchaseDtos.PurchaseCategory;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
import java.util.UUID;

/** Stats screen — donut by category + by day. */
public final class StatsDtos {

    private StatsDtos() {}

    public record CategorySpendDto(
            PurchaseCategory category,
            MoneyDto amount,
            /** 0..1 share of total. */
            BigDecimal share,
            int purchaseCount
    ) {}

    public record DailySpendDto(
            LocalDate date,
            MoneyDto amount,
            int purchaseCount
    ) {}

    public record TripStatsDto(
            UUID tripId,
            MoneyDto totalSpent,
            MoneyDto budget,
            MoneyDto dailyAverage,
            MoneyDto remaining,
            boolean onBudget,
            /** Positive days early/late vs linear burn; null if no dates. */
            Integer paceDeltaDays,
            PurchaseCategory topCategory,
            List<CategorySpendDto> byCategory,
            List<DailySpendDto> byDay
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/data/remote/dto/stats/StatsDtos.kt
package com.example.shoptourr.data.remote.dto.stats

import com.example.shoptourr.data.remote.dto.common.MoneyDto
import com.example.shoptourr.data.remote.dto.purchase.PurchaseCategory

data class CategorySpendDto(
    val category: PurchaseCategory,
    val amount: MoneyDto,
    val share: String,
    val purchaseCount: Int,
)

data class DailySpendDto(
    val date: String,
    val amount: MoneyDto,
    val purchaseCount: Int,
)

data class TripStatsDto(
    val tripId: String,
    val totalSpent: MoneyDto,
    val budget: MoneyDto,
    val dailyAverage: MoneyDto,
    val remaining: MoneyDto,
    val onBudget: Boolean,
    val paceDeltaDays: Int? = null,
    val topCategory: PurchaseCategory? = null,
    val byCategory: List<CategorySpendDto>,
    val byDay: List<DailySpendDto>,
)
